projects/plugins/jetpack/extensions/blocks/videopress/edit.js

```diff
@@ -402,28 +402,34 @@ const VideoPressEdit = CoreVideoEdit =>
 				</Fragment>
 			);
 
-			if ( isUploading ) {
+			const isFetchingVideo = isFetchingMedia || isFetchingPreview;
+			const isVideoFallbackOrNotPreview = fallback || ! preview;
+			const displayCoreVideoBlock =
+				isVideoFallbackOrNotPreview && ! isUploading && ! isFetchingVideo;
+			const renderCoreVideoAndLoadingBlocks =
+				isUploading || isFetchingVideo || displayCoreVideoBlock;
```

Comment:
Do you think you could add some comments around here, so this can be a bit easier to parse for anyone of us coming to this file in a few months? I think that would help, as the logic here is getting a bit tricky!

Reply:
Sure, I'll update this :)

Reply:
✅ I both added comments and simplified/aggregated a condition